app/forms/declaration/WarehouseIdentification.scala

```diff
@@ -57,8 +57,16 @@ object WarehouseIdentification {
           )
       )
     )(WarehouseIdentification.apply)(WarehouseIdentification.unapply)
+    .verifying("supplementary.warehouse.identificationType.error", typeSelectedWhenNumberIsPopulated)
     .verifying("supplementary.warehouse.identificationNumber.empty", idNumberIsPopulatedWhenIDTypeIsSelected)
 
+  private def typeSelectedWhenNumberIsPopulated: WarehouseIdentification => Boolean =
+    w =>
```

Comment:
I know that the scope of this `w` is small, but I think it's really better to use this `warehouse` or something different. In few places, where you have bigger part of the code or a lot of one letter variables it's time consuming to connect this with correct name